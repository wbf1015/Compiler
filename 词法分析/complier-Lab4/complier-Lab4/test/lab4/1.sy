int a;

int main()
{
    int a;
    a = 1 + 22.2 + 047 + 0x10;
    if(a < 5)
        return 1;
    return 0;
}