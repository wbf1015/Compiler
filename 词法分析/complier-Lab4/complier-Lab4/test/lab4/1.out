[DEBUG LAB4]: 	INT		int	1	0
[DEBUG LAB4]: 	ID		a	1	4	93957663265664
[DEBUG LAB4]: 	SEMICOLON		;	1	5
[DEBUG LAB4]: 	INT		int	3	0
[DEBUG LAB4]: 	ID		main	3	4	93957663266048
[DEBUG LAB4]: 	LPAREN		(	3	8
[DEBUG LAB4]: 	RPAREN		)	3	9
[DEBUG LAB4]: 	LBRACE		{	4	0
[DEBUG LAB4]: 	INT		int	5	4
[DEBUG LAB4]: 	ID		a	5	8	93957663266144
[DEBUG LAB4]: 	SEMICOLON		;	5	9
[DEBUG LAB4]: 	ID		a	6	4	93957663266144
[DEBUG LAB4]: 	ASSIGN		=	6	6
[DEBUG LAB4]: 	DECIMIAL_INT		1	6	8	1.000000
[DEBUG LAB4]: 	ADD		+	6	10
[DEBUG LAB4]: 	DECIMIAL_FLOAT		22.2	6	12	22.200001
[DEBUG LAB4]: 	ADD		+	6	17
[DEBUG LAB4]: 	OCT		047	6	19	39.000000
[DEBUG LAB4]: 	ADD		+	6	23
[DEBUG LAB4]: 	HEX		0x10	6	25	16.000000
[DEBUG LAB4]: 	SEMICOLON		;	6	29
[DEBUG LAB4]: 	IF		if	7	4
[DEBUG LAB4]: 	LPAREN		(	7	6
[DEBUG LAB4]: 	ID		a	7	7	93957663266144
[DEBUG LAB4]: 	LESS		<	7	9
[DEBUG LAB4]: 	DECIMIAL_INT		5	7	11	5.000000
[DEBUG LAB4]: 	RPAREN		)	7	12
[DEBUG LAB4]: 	RETURN		return	8	8
[DEBUG LAB4]: 	DECIMIAL_INT		1	8	15	1.000000
[DEBUG LAB4]: 	SEMICOLON		;	8	16
[DEBUG LAB4]: 	RETURN		return	9	4
[DEBUG LAB4]: 	DECIMIAL_INT		0	9	11	0.000000
[DEBUG LAB4]: 	SEMICOLON		;	9	12
[DEBUG LAB4]: 	RBRACE		}	10	0
